### ISSUE TITLE
Feature Request: control vline width so no gaps/overlap between candles

### ISSUE THREAD

**Issue by twopirllc** [matplotlib/mplfinance#240]
Hello @DanielGoldfarb,

**Is your feature request related to a problem? Please describe.**
I like the _vlines_ but they sometimes overlap when ```linewidths >= 8``` or create gaps when ```linewidths < 8```. It's noticeable when the number of bars are plotted.

**Describe the solution you'd like**
* Either another argument to passed into _vlines_ so that there are no gaps or overlaps created.
* Or maybe something similar to _vlines_ that is more specific for this purpose as this feature is slightly different than the original purpose of _vlines_.

**Describe alternatives you've considered**
I have not checked if it works in **External Axes Mode**.

**Additional context**
I made a semi-dynamic Panels [Chart Class Example Notebook](https://github.com/twopirllc/pandas-ta/blob/master/examples/example.ipynb) for my **Pandas TA** Library. The _plot_ method is located at the bottom of _Input Cell 11_.

You can play around with the Chart class arguments, namely _last_ argument,  in _Input Cell 12_ to see what I mean. 

Here are some example chart outputs:
#### Gaps
```last = 30``` and no _linewidth_ argument used.
<img width="673" alt="Screen Shot 2020-08-15 at 11 18 00 AM" src="https://user-images.githubusercontent.com/18198392/90318978-10462480-dee9-11ea-9329-d66b022c8bc5.png">

```last = 90``` and no _linewidth_ argument used.
<img width="655" alt="Screen Shot 2020-08-15 at 11 22 58 AM" src="https://user-images.githubusercontent.com/18198392/90319074-bd20a180-dee9-11ea-9b27-c480cd20177c.png">

```last = recent``` (recent = 251) and no _linewidth_ argument used.
<img width="658" alt="Screen Shot 2020-08-15 at 11 24 27 AM" src="https://user-images.githubusercontent.com/18198392/90319093-e8a38c00-dee9-11ea-8b55-42d4423c8fea.png">

```last = 2 * recent``` (recent = 502) and no _linewidth_ argument used.
<img width="675" alt="Screen Shot 2020-08-15 at 11 25 12 AM" src="https://user-images.githubusercontent.com/18198392/90319107-0cff6880-deea-11ea-96cc-ffc82eb42459.png">

```last = 30, linewidths=8```
<img width="639" alt="Screen Shot 2020-08-15 at 11 35 04 AM" src="https://user-images.githubusercontent.com/18198392/90319272-64520880-deeb-11ea-89b0-c25aad00a2a4.png">

#### Overlap
```last = 90, linewidths=8```
<img width="654" alt="Screen Shot 2020-08-15 at 11 35 56 AM" src="https://user-images.githubusercontent.com/18198392/90319285-82b80400-deeb-11ea-95f6-1e1ba2f3258d.png">

```last = recent, linewidths=8```
<img width="676" alt="Screen Shot 2020-08-15 at 11 37 03 AM" src="https://user-images.githubusercontent.com/18198392/90319301-ac712b00-deeb-11ea-8b31-d91f804eb3a2.png">

Hope this helps!

Thanks for your time,
Kevin

**Comment by DanielGoldfarb** [comment]
Hi Kevin,
Need a  little clarification on a couple things.  

1. When you say "linewidths" are you referring to passing in a `linewidths` key as part of the `vlines` `dict`?  (which I don't see in your code, but maybe your are saying you had it in the code when you ran some of the above examples?).  Or do you mean something else by "linewidths" in the above descriptions.

2. What are you trying to accomplish with your charts? <br>  What is the meaning/significance of `amat_sr`? <br> Perhaps your suggestion `"Or maybe something ... more specific for this purpose ..."` is what is needed here,  but I need to better understand what you mean by `"this purpose"`.  <br>How exactly do you want the vertical lines in the charts to look?<br>   **For example, what would you change** about the appearance of [**this chart**](https://user-images.githubusercontent.com/18198392/90319272-64520880-deeb-11ea-89b0-c25aad00a2a4.png)?

Thanks.  --Daniel

**Comment by twopirllc** [comment]
Hello Daniel,

1. Above I meant ```vlines``` _key_ and not argument; sorry for the confusion. I used the ```linewidths``` key for testing but removed it prior to pushing the final version to my repo.

2. So ```amat_sr``` is the bearish result of my **amat** indicator. Ideally, there would be no white space between the three pink ```vlines``` on the example chart you linked, it would be one continuous block of pink, no gaps and fully enclosing the candles. Like this chart (but imagine the orange one is also pink):

<img width="735" alt="Screen Shot 2020-08-16 at 8 20 10 AM" src="https://user-images.githubusercontent.com/18198392/90337912-83f43a00-df9a-11ea-864a-c3f94d3a119c.png">

3. Ultimately, it would be great to pass an array dicts to ```vline``` to have multiple colors for different conditions like the pink and orange backgrounds in the main price panel below.

```python
red_dates = red_condition_df.index.to_list()
orange_dates = orange_condition_df.index.to_list()
vlines = [
    dict(vlines= red_dates, colors= 'red', alpha= 0.1),
    dict(vlines= orange_dates, colors= 'orange', alpha= 0.1)
]
```

<img width="940" alt="Screen Shot 2020-08-16 at 8 31 03 AM" src="https://user-images.githubusercontent.com/18198392/90338008-5f4c9200-df9b-11ea-9e49-be4a211c1b37.png">

Hope this makes more sense.

Thanks,
Kevin

**Comment by DanielGoldfarb** [comment]
Hmmm.  Will have to think this one through a bit.  Possibly some version of `fill_between` will work (see the bottom third, below cell [10], of [the customizations notebook](https://github.com/matplotlib/mplfinance/blob/master/examples/plot_customizations.ipynb)) 

I imagine `vlines` is very convenient for what you are trying to accomplish, since using `vlines` works nicely with looping through your data checking your indicator.  Still, play around with `fill_between` and see if that works for you.  

Meanwhile I will think about this some more.  Part of the problem with using linewidths is that the meaning of a given width can vary depending on many factors, including how much data there is on the chart, how large the figure is, etc.  There is an algorithim in mplfinance that attempts to adjust, for example, candle widths, depending on these factors, so as to try to get nice looking widths most of the time.  In a sense, you are trying to coordinate your `vline` widths with the candle widths, but the algorithm is a black box from your perspective.  Thus, the need to think this through a little more.  Meanwhile try `fill_between` and let me know if that works or if `fill_between` also has some issues.  If you want to read about the width algorithm you can do so [**here**](https://github.com/matplotlib/mplfinance/blob/master/examples/widths.ipynb).

**Comment by twopirllc** [comment]
Yes, the ```vlines``` have been convenient. I will read up on the _width_ algorithm you linked.

Thank you for the suggestion to use ```fill_between```. I will play around with it. I also plan to use ```fill_between``` with Bollinger Bands, Keltner channels, et al.

Thank you for your time,
Kevin

**Comment by DanielGoldfarb** [comment]
just want to share [**a `fill_between` example**](https://user-images.githubusercontent.com/11164790/76273899-70a7aa80-6255-11ea-8636-f2cff10b3ba8.jpg) that I made a long time ago (just prior to getting involved with mplfinance).
